packages/server/api/controllers/products.controller.js

```diff
@@ -33,22 +33,43 @@ const getProductById = async (id) => {
 };
 
 // Get products by Category
-const getProductsByCategory = async (category, limit, offset) => {
+const getProductsByCategory = async (token, category, limit, offset) => {
+  let products = [];
   if (!isNaN(category)) {
     throw new HttpError('Category should be a String', 400);
   }
-
   try {
-    const products = await knex('products')
-      .select(
-        'products.id as id',
-        'products.name',
-        'products.description',
-        'products.price',
-        'categories.name as cname',
-      )
-      .leftJoin('categories', 'products.category_id', 'categories.id')
-      .offset(offset);
+    if (token) {
+      const userUid = token.split(' ')[1];
+      const user = (await knex('users').where({ uid: userUid }))[0];
+      products = await knex('products')
+        .select(
+          'products.id as id',
+          'products.name',
+          'products.description',
+          'products.price',
+          'categories.name as cname',
+          'favorites.id as favoritesID',
+        )
+        .leftJoin('categories', 'products.category_id', 'categories.id')
+        .offset(offset)
+        .leftJoin('favorites', function () {
+          this.on('products.id', '=', 'favorites.product_id');
+          this.andOnVal('favorites.user_id', '=', `${user ? user.id : ''}`);
+        });
+    } else {
+      products = await knex('products')
+        .select(
+          'products.id as id',
+          'products.name',
+          'products.description',
+          'products.price',
+          'categories.name as cname',
```

Comment:
I have tried to add NULL as favoritesID as an extra column but Its not working.  Its working in sql query but not accepting null value in knex select query.

Reply:
My sleep-deprived understanding:
We never fall into else, as `token` is always true, because this is how we pass it:  
<img width="383" alt="image" src="https://user-images.githubusercontent.com/64137994/213014648-8dc03a6b-df93-4db0-80da-6e457baae7f3.png">

So, we always have a list where every product includes a `favoritesID`, event when not logged in:
![image](https://user-images.githubusercontent.com/64137994/213014733-0aa65c5c-deeb-4ddc-a956-5597c00e526e.png)

Therefore, in `ProductList`, this checks out as false:
<img width="477" alt="image" src="https://user-images.githubusercontent.com/64137994/213015030-50f10dba-048c-40f3-98df-772bd83ad287.png">

And we have no favourites in any category. 
Let's merge! :D

Reply:
I agree, let's remove the else part.